Update reviewing and merging documentation

brentswisher opened the pull request:
The list of open pull requests for Gutenberg keeps growing. This is an attempt to update the developer documentation to encourage more people to participate in the reviewing and merging process, as discussed in the recent developer meeting - https://wordpress.slack.com/archives/C02QB2JS7/p1564580809002700

brentswisher committed:
Update reviewing and merging documentation to clarify process and encourage newer contributors

tellthemachines commented on docs/contributors/repository-management.md:
```diff
@@ -116,6 +116,8 @@ Every pull request goes through a manual code review, in addition to automated t
 *As a contributor*, your responsibility is to learn from suggestions and iterate your pull request should it be needed based on feedback. Seek to collaborate and produce the best possible contribution to the greater whole.
 
+Code reviews are encouraged by everyone who is willing to attempt one. If you review a pull request and are confident in the changes, approve it. If you don't feel totally confident it is ready for merging, add your review with a comment that says it should have another set of eyes on it before final approval. This can help filter out obvious bugs and simplify reviews for core members. It also is ...
+
```
Might be good to add that it's also fine to just comment with questions when you're unsure of how something should work or if the reason for the change is unclear. Or review only the parts of the code you are comfortable with, and comment that changes x and y haven't been looked at yet. 
Great to emphasise code review as a learning experience 🙂

brentswisher replied:
Thanks for the feedback! I took a stab at what you were suggesting, let me know what you think.

tellthemachines replied:
Looks great, thanks for this!

gziolo commented on docs/contributors/repository-management.md:
```diff
@@ -138,7 +140,10 @@ A pull request can generally be merged once it is:
 The final pull request merge decision is made by the **@wordpress/gutenberg-core** team.
 
-Please make sure to assign your merged pull request to its release milestone. Doing so creates the historical legacy of what code landed when, and makes it possible for all project contributors (even non-technical ones) to access this information.
+All members of the Wordpress organization on Github have the ability to review and merge pull requests. If you have reviewed a PR and are confident in the code, approve the pull request and comment pinging **@wordpress/gutenberg-core** or a specific core member who has been involved in the PR. Once they confirm there are no objections, you are free to merge the PR into master.
```
Typos:
`Wordpress` -> `WordPress`
`Github` -> `GitHub`

brentswisher replied:
Thanks! Updated

jasmussen commented:
With the changes suggested already, this seems good to me 👍 👍

mapk committed:
Updated  a sentence for clarity

The sentence didn't sound quite right, so I hope you don't mind the update. If it's at all not what you wanted, feel free to revert it.

brentswisher committed:
Fix typos in updated documentation, including that dang capital "P"

brentswisher committed:
Add additional sentence to pull request documentation encouraging comments if someone is not comfortable leaving a full review as suggested by @tellthemachines

mapk reviewed (approved):
:shipit:
